From 096a4e6e002c4cbd9cae7bf49a27704756301831 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.liu@freescale.com>
Date: Thu, 10 Jan 2013 16:32:43 +0800
Subject: [PATCH 385/505] ENGR00239598 V4L2:Support several tests for Android

This patch supports the following unit tests for Android:
mxc_v4l2_capture
mxc_v4l2_overlay
mxc_v4l2_output
mxc_v4l2_tvin

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 test/mxc_v4l2_test/Android.mk         | 41 +++++++++++++++++++++++++++++++++++
 test/mxc_v4l2_test/mxc_v4l2_output.c  |  6 ++++-
 test/mxc_v4l2_test/mxc_v4l2_overlay.c | 16 ++++++++++++++
 test/mxc_v4l2_test/mxc_v4l2_tvin.c    |  6 ++++-
 4 files changed, 67 insertions(+), 2 deletions(-)
 create mode 100644 test/mxc_v4l2_test/Android.mk

diff --git a/test/mxc_v4l2_test/Android.mk b/test/mxc_v4l2_test/Android.mk
new file mode 100644
index 0000000..8346f29
--- /dev/null
+++ b/test/mxc_v4l2_test/Android.mk
@@ -0,0 +1,41 @@
+ifeq ($(HAVE_FSL_IMX_IPU),true)
+
+LOCAL_PATH := $(call my-dir)
+
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := mxc_v4l2_capture.c
+LOCAL_CFLAGS += -DBUILD_FOR_ANDROID
+LOCAL_C_INCLUDES += $(LOCAL_PATH)
+LOCAL_SHARED_LIBRARIES := libutils libc
+LOCAL_MODULE := mxc-v4l2-capture
+LOCAL_MODULE_TAGS := tests
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := mxc_v4l2_overlay.c
+LOCAL_CFLAGS += -DBUILD_FOR_ANDROID
+LOCAL_C_INCLUDES += $(LOCAL_PATH)
+LOCAL_SHARED_LIBRARIES := libutils libc
+LOCAL_MODULE := mxc-v4l2-overlay
+LOCAL_MODULE_TAGS := tests
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := mxc_v4l2_output.c
+LOCAL_CFLAGS += -DBUILD_FOR_ANDROID
+LOCAL_C_INCLUDES += $(LOCAL_PATH)
+LOCAL_SHARED_LIBRARIES := libutils libc
+LOCAL_MODULE := mxc-v4l2-output
+LOCAL_MODULE_TAGS := tests
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := mxc_v4l2_tvin.c
+LOCAL_CFLAGS += -DBUILD_FOR_ANDROID
+LOCAL_C_INCLUDES += $(LOCAL_PATH)
+LOCAL_SHARED_LIBRARIES := libutils libc
+LOCAL_MODULE := mxc-v4l2-tvin
+LOCAL_MODULE_TAGS := tests
+include $(BUILD_EXECUTABLE)
+
+endif
diff --git a/test/mxc_v4l2_test/mxc_v4l2_output.c b/test/mxc_v4l2_test/mxc_v4l2_output.c
index 224e257..ddf6513 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_output.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_output.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2012 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright 2004-2013 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -94,7 +94,11 @@ void fb_setup(void)
         struct mxcfb_gbl_alpha alpha;
 	int fd;
 
+#ifdef BUILD_FOR_ANDROID
 	fd = open("/dev/fb0",O_RDWR);
+#else
+	fd = open("/dev/graphics/fb0",O_RDWR);
+#endif
 
         alpha.alpha = 0;
         alpha.enable = 1;
diff --git a/test/mxc_v4l2_test/mxc_v4l2_overlay.c b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
index 1c8e491..8f609c1 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_overlay.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
@@ -132,7 +132,11 @@ mxc_v4l_overlay_test(int timeout)
         int overlay = 1;
         int retval = 0;
         struct v4l2_control ctl;
+#ifdef BUILD_FOR_ANDROID
+	char fb_device_0[100] = "/dev/graphics/fb0";
+#else
 	char fb_device_0[100] = "/dev/fb0";
+#endif
 	int fd_graphic_fb = 0;
 	struct fb_var_screeninfo fb0_var;
 	struct mxcfb_loc_alpha l_alpha;
@@ -580,7 +584,11 @@ main(int argc, char **argv)
 {
         struct v4l2_format fmt;
         struct v4l2_framebuffer fb_v4l2;
+#ifdef BUILD_FOR_ANDROID
+        char fb_device_0[100] = "/dev/graphics/fb0";
+#else
         char fb_device_0[100] = "/dev/fb0";
+#endif
         char *fb_device_fg;
         int fd_fb_0 = 0;
         struct fb_fix_screeninfo fb0_fix, fb_fg_fix;
@@ -661,7 +669,11 @@ main(int argc, char **argv)
 
         memset(&fb_v4l2, 0, sizeof(fb_v4l2));
 
+#ifdef BUILD_FOR_ANDROID
+	fb_device_fg = "/dev/graphics/fb1";
+#else
 	fb_device_fg = "/dev/fb1";
+#endif
 	if ((g_fd_fb_fg = open(fb_device_fg, O_RDWR)) < 0) {
 		printf("Unable to open frame buffer 1\n");
 		close(fd_fb_0);
@@ -678,7 +690,11 @@ main(int argc, char **argv)
 	if (((strcmp(fb_fg_fix.id, "DISP3 FG") != 0) && g_display_lcd < 3)
 		|| ((strcmp(fb_fg_fix.id, "DISP4 FG") != 0) && g_display_lcd > 2)) {
 		close(g_fd_fb_fg);
+#ifdef BUILD_FOR_ANDROID
+		fb_device_fg = "/dev/graphics/fb2";
+#else
 		fb_device_fg = "/dev/fb2";
+#endif
 		if ((g_fd_fb_fg = open(fb_device_fg, O_RDWR)) < 0) {
 			printf("Unable to open frame buffer 2\n");
 			close(fd_fb_0);
diff --git a/test/mxc_v4l2_test/mxc_v4l2_tvin.c b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
index d47b4ab..06b1437 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_tvin.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2007-2012 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright 2007-2013 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -667,7 +667,11 @@ int process_cmdline(int argc, char **argv)
 
 int main(int argc, char **argv)
 {
+#ifdef BUILD_FOR_ANDROID
+	char fb_device[100] = "/dev/graphics/fb0";
+#else
 	char fb_device[100] = "/dev/fb0";
+#endif
 	int fd_fb = 0, i;
 	struct mxcfb_gbl_alpha alpha;
 	enum v4l2_buf_type type;
-- 
1.8.0

